JobWorker does not receive variables when using @JsonProperty

**Describe the bug**

I am using a Job Worker that receives Process Variables as a DTO, mapped via `@VariablesAsType` annotation. The problem is that the DTOs fields become `null` (i.e. in case of Kotlin it fails for non-nullable fields) as soon as I map field names using `@JsonProperty`

Here is the Worker and the DTO:

```
@Component
class DoSomethingJobWorker(
) {

    @JobWorker(type = "do_something")
    fun handle(@VariablesAsType dto: DoSomethingDto) {
        return
    }
}

data class DoSomethingDto(
    @JsonProperty("some_value")
    val someValue: String,
)
```

Interestingly in each of the following variations, the variable gets correctly set into the DTO:

1) injecting activated job:

```
fun handle(@VariablesAsType dto: DoSomethingDto, activatedJob: ActivatedJob) {
```

2) Setting fetchAllVariables to true

```
@JobWorker(type = "do_something", fetchAllVariables = true)
```

3) not using `@JsonProperty`

```
data class DoSomethingDto(
    val some_value: String,
)
```
It is enought to use one of the adjustments (1, 2 or 3) to make it work

**To Reproduce**

- Create a simple process "Start" -> "Service Task" -> "End"
- Define the above Job Worker in a Spring application
- Start the process with { "some_value": "value" }


**Expected behavior**

The value passed to the process should be present in the DTO when the original Job Worker (without the mentioned adjustments) is executed. It does not make any sense to me, why e.g. injecting the activated job "fixes" the issue.


**Log/Stacktrace**

<!-- If possible add the full stacktrace or log which contains the issue. -->

<details><summary>Full Stacktrace</summary>
 <p>

```
io.camunda.zeebe.client.api.command.InternalClientException: Failed to deserialize json '{}' to class 'class com.example.DoSomethingDto'
	at io.camunda.zeebe.client.impl.ZeebeObjectMapper.fromJson(ZeebeObjectMapper.java:56) ~[zeebe-client-java-8.5.5.jar:8.5.5]
	at io.camunda.zeebe.client.impl.response.ActivatedJobImpl.getVariablesAsType(ActivatedJobImpl.java:146) ~[zeebe-client-java-8.5.5.jar:8.5.5]
	at io.camunda.zeebe.spring.client.jobhandling.parameter.VariablesAsTypeResolver.resolve(VariablesAsTypeResolver.java:15) ~[spring-client-zeebe-8.5.9.jar:8.5.9]
	at io.camunda.zeebe.spring.client.jobhandling.JobHandlerInvokingSpringBeans.lambda$createParameters$0(JobHandlerInvokingSpringBeans.java:98) ~[spring-client-zeebe-8.5.9.jar:8.5.9]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[na:na]
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:722) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:575) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluateToArrayNode(AbstractPipeline.java:260) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:616) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.toArray(ReferencePipeline.java:622) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.toList(ReferencePipeline.java:627) ~[na:na]
	at io.camunda.zeebe.spring.client.jobhandling.JobHandlerInvokingSpringBeans.createParameters(JobHandlerInvokingSpringBeans.java:98) ~[spring-client-zeebe-8.5.9.jar:8.5.9]
	at io.camunda.zeebe.spring.client.jobhandling.JobHandlerInvokingSpringBeans.handle(JobHandlerInvokingSpringBeans.java:52) ~[spring-client-zeebe-8.5.9.jar:8.5.9]
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactoryImpl.executeJob(JobRunnableFactoryImpl.java:45) ~[zeebe-client-java-8.5.5.jar:8.5.5]
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactoryImpl.lambda$create$0(JobRunnableFactoryImpl.java:40) ~[zeebe-client-java-8.5.5.jar:8.5.5]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:572) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:317) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:1583) ~[na:na]
Caused by: com.fasterxml.jackson.module.kotlin.MissingKotlinParameterException: Instantiation of [simple type, class com.example.DoSomethingDto] value failed for JSON property some_value due to missing (therefore NULL) value for creator parameter someValue which is a non-nullable type
 at [Source: REDACTED (`StreamReadFeature.INCLUDE_SOURCE_IN_LOCATION` disabled); line: 1, column: 2] (through reference chain: com.example.DoSomethingDto["some_value"])
	at com.fasterxml.jackson.module.kotlin.KotlinValueInstantiator.createFromObjectWith(KotlinValueInstantiator.kt:97) ~[jackson-module-kotlin-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.deser.impl.PropertyBasedCreator.build(PropertyBasedCreator.java:202) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer._deserializeUsingPropertyBased(BeanDeserializer.java:532) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializerBase.deserializeFromObjectUsingNonDefault(BeanDeserializerBase.java:1493) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserializeFromObject(BeanDeserializer.java:348) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.deser.BeanDeserializer.deserialize(BeanDeserializer.java:185) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.deser.DefaultDeserializationContext.readRootValue(DefaultDeserializationContext.java:342) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.ObjectMapper._readMapAndClose(ObjectMapper.java:4905) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3848) ~[jackson-databind-2.17.2.jar:2.17.2]
	at com.fasterxml.jackson.databind.ObjectMapper.readValue(ObjectMapper.java:3816) ~[jackson-databind-2.17.2.jar:2.17.2]
	at io.camunda.zeebe.client.impl.ZeebeObjectMapper.fromJson(ZeebeObjectMapper.java:53) ~[zeebe-client-java-8.5.5.jar:8.5.5]
	... 23 common frames omitted
```

</p>
</details>

**Environment:**
- OS: Linux, MacOS
- Deployment: Self Managed Camunda 8
- Spring-Zeebe Version: 8.5.5
- SpringBoot Version: 3.3.2

--- a/spring-boot-starter-camunda/pom.xml
+++ b/spring-boot-starter-camunda/pom.xml
@@ -45,6 +45,11 @@
       <scope>provided</scope>
     </dependency>
     <dependency>
+      <groupId>com.fasterxml.jackson.core</groupId>
+      <artifactId>jackson-annotations</artifactId>
+      <scope>provided</scope>
+    </dependency>
+    <dependency>
       <groupId>io.camunda.spring</groupId>
       <artifactId>spring-client-zeebe</artifactId>
     </dependency>

--- a/spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java
+++ b/spring-boot-starter-camunda/src/main/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizer.java
@@ -1,10 +1,10 @@
 package io.camunda.zeebe.spring.client.properties;
 
 import static io.camunda.zeebe.spring.client.configuration.PropertyUtil.*;
-import static io.camunda.zeebe.spring.client.properties.ZeebeClientConfigurationProperties.*;
 import static java.util.Optional.*;
 import static org.apache.commons.lang3.StringUtils.*;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import io.camunda.zeebe.client.api.response.ActivatedJob;
 import io.camunda.zeebe.spring.client.annotation.Variable;
 import io.camunda.zeebe.spring.client.annotation.VariablesAsType;
@@ -16,6 +16,7 @@ import io.camunda.zeebe.spring.client.bean.CopyNotNullBeanUtilsBean;
 import io.camunda.zeebe.spring.client.bean.MethodInfo;
 import io.camunda.zeebe.spring.client.bean.ParameterInfo;
 import io.camunda.zeebe.spring.client.properties.common.ZeebeClientProperties;
+import java.lang.reflect.Field;
 import java.lang.reflect.InvocationTargetException;
 import java.util.ArrayList;
 import java.util.HashMap;
@@ -104,8 +105,15 @@ public class PropertyBasedZeebeWorkerValueCustomizer implements ZeebeWorkerValue
     parameters.forEach(
         pi ->
             ReflectionUtils.doWithFields(
-                pi.getParameterInfo().getType(), f -> result.add(f.getName())));
+                pi.getParameterInfo().getType(), f -> result.add(extractParameterName(f))));
     return result;
+  }
+
+  private String extractParameterName(Field field) {
+    if (field.isAnnotationPresent(JsonProperty.class)) {
+      return field.getAnnotation(JsonProperty.class).value();
+    }
+    return field.getName();
   }
 
   private void applyOverrides(ZeebeWorkerValue zeebeWorker) {

--- a/spring-boot-starter-camunda/src/test/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizerTest.java
+++ b/spring-boot-starter-camunda/src/test/java/io/camunda/zeebe/spring/client/properties/PropertyBasedZeebeWorkerValueCustomizerTest.java
@@ -2,6 +2,7 @@ package io.camunda.zeebe.spring.client.properties;
 
 import static org.assertj.core.api.Assertions.*;
 
+import com.fasterxml.jackson.annotation.JsonProperty;
 import io.camunda.zeebe.client.api.response.ActivatedJob;
 import io.camunda.zeebe.spring.client.annotation.JobWorker;
 import io.camunda.zeebe.spring.client.annotation.Variable;
@@ -46,6 +47,9 @@ public class PropertyBasedZeebeWorkerValueCustomizerTest {
 
   @JobWorker
   void sampleWorker(@Variable String var1, @VariablesAsType ComplexProcessVariable var2) {}
+
+  @JobWorker
+  void sampleWorkerWithJsonProperty(@VariablesAsType PropertyAnnotatedClass var2) {}
 
   @JobWorker
   void activatedJobWorker(@Variable String var1, ActivatedJob activatedJob) {}
@@ -360,6 +364,19 @@ public class PropertyBasedZeebeWorkerValueCustomizerTest {
     assertThat(zeebeWorkerValue.getEnabled()).isFalse();
   }
 
+  @Test
+  void shouldApplyPropertyAnnotationOnVariableFiltering() {
+    // given
+    PropertyBasedZeebeWorkerValueCustomizer customizer =
+        new PropertyBasedZeebeWorkerValueCustomizer(legacyProperties(), properties());
+    ZeebeWorkerValue zeebeWorkerValue = new ZeebeWorkerValue();
+    zeebeWorkerValue.setMethodInfo(methodInfo(this, "testBean", "sampleWorkerWithJsonProperty"));
+    // when
+    customizer.customize(zeebeWorkerValue);
+    // then
+    assertThat(zeebeWorkerValue.getFetchVariables()).containsExactly("weird_name");
+  }
+
   private static class ComplexProcessVariable {
     private String var3;
     private String var4;
@@ -379,5 +396,10 @@ public class PropertyBasedZeebeWorkerValueCustomizerTest {
     public void setVar4(String var4) {
       this.var4 = var4;
     }
+  }
+
+  private static class PropertyAnnotatedClass {
+    @JsonProperty("weird_name")
+    private String value;
   }
 }

